meta-resin-common/recipes-containers/docker-disk/docker-resin-supervisor-disk/update-resin-supervisor

```diff
@@ -133,24 +124,32 @@ else
     tag=$UPDATER_SUPERVISOR_TAG
 fi
 
-# Try to stop old supervisor to prevent from deleting the intermediate images while downloading the new one
-echo "Stop supervisor..."
-systemctl stop resin-supervisor
-
-# Store the tagged image string so resin-supervisor.service can pick it up
-echo "SUPERVISOR_IMAGE=$image_name:$tag" > $UPDATECONF
+# Get image id of tag. This will be non-empty only in case it's already downloaded.
+echo "Getting image id..."
+imageid=$($DOCKER inspect -f '{{.Id}}' "$image_name:$tag") || imageid=""
 
-# Pull target version.
-echo "Update supervisor..."
-$DOCKER pull "$image_name:$tag"
-
-# Run supervisor with the device-type-specific options.
-# We give a specific name to the container to guarantee only one running.
-echo "Start supervisor..."
-systemctl start resin-supervisor
+if [ -n "$imageid" ]; then
+    echo "Supervisor $image_name:$tag already downloaded."
+    exit 0
+else
```

Comment:
You don't need all this in else. You can just let the code flow if the check of imageid didn't exit.